resources/assets/lib/utils/beatmap-helper.ts

```diff
@@ -82,6 +88,11 @@ export function getDiffRating(rating: number) {
   return 'expert-plus';
 }
 
+export function getDiffColour(rating?: number | null) {
+  if (rating == null || rating < 0 || rating > 8) return '#000000';
```

Comment:
Pretty sure null/0 should not be the same as >8. If anything it should default to the lowest colour value.